File: js/examples/live-trades.ts
// Live trades: subscribe to the `trades` channel (incremental — one message per new print).
// Run:  node --experimental-strip-types examples/live-trades.ts
import { ColibriClient, type TradePush } from "../src/index.js";

const client = await ColibriClient.discover();
const ws = client.stream();

ws.on("trades", (data) => {
  const p = data as { trades: TradePush[] };
  for (const t of p.trades) {
    const side = t.isBuy ? "BUY " : "SELL";
    console.log(`${side}  ${t.qty} @ ${t.price}`);
  }
});

await ws.connect();
ws.subscribe("trades", { exchange: "BinanceSpot", symbol: "BTCUSDT" });

console.log("streaming BTCUSDT trades — Ctrl+C to stop");

File: python/examples/live_trades.py
"""Live trades: subscribe to the `trades` channel and print each print.

Requires websocket-client:  pip install websocket-client

    python examples/live_trades.py
"""
from colibri import ColibriClient

client = ColibriClient.discover()
ws = client.stream()


def on_trades(data, _frame):
    for t in data.get("trades", []):
        side = "BUY " if t["isBuy"] else "SELL"
        print(f"{side}  {t['qty']} @ {t['price']}")


ws.on("trades", on_trades)
ws.connect()
ws.subscribe("trades", exchange="BinanceSpot", symbol="BTCUSDT")

print("streaming BTCUSDT trades — Ctrl+C to stop")
ws.run_forever()
